src/adapter/pg.cr

```diff
@@ -76,6 +76,47 @@ class Granite::Adapter::Pg < Granite::Adapter::Base
     end
   end
 
+  def import(table_name : String, primary_name : String, fields, model_array, **options)
```

Comment:
your `**options` can be a little cleaner and have a similar effect:

```crystal
def import(table_name : String, primary_name : String, fields, model_array : Array(self), *, update_on_duplicate_key = false)
```

Reply:
That would mean i would need to have a defined param for each options right?

As of now there is `on_duplicate_key_ignore: Bool`, `on_duplicate_key_update: Array(Strings) to be updated`, and possibly in future also `validate: true`, as giving option to ignore validation might speed up large imports.

Reply:
You can probably have the best of both worlds using a crystal overloaded method signature. If you need, here are [the docs on overloading](https://crystal-lang.org/docs/syntax_and_semantics/overloading.html)

Reply:
As it happens, I don’t have strong opinions on this either. I just noticed a rubyism that might be able to be cleaned up with some crystal.